src/server/snapshot.cc

```diff
@@ -272,24 +272,36 @@ void SliceSnapshot::OnDbChange(DbIndex db_index, const DbSlice::ChangeReq& req)
   }
 }
 
+// For any key any journal entry must arrive at the replica strictly after its first original rdb
+// value.
+// TODO: We should skip it if on db change was called - possible bug!
+// For example: l =  {v1, v2}, LPUSH l v3 writes {v1, v2, v3} in rdb, and sends LPUSH l v3
+// -> duplicate l3
+// Soltuion: Compare snapshot versions?
+//
+// TODO: Possible bug when value is in default buffer and we switch db, then receive journal event
+// and write it to tmp serializer and flush it.
+// Soltion: Impossible because we force flush after iterating the db?
 void SliceSnapshot::OnJournalEntry(const journal::Entry& entry) {
```

Comment:
Some notes about the journal change listener in snapshot. We discussed both of them, but after a second thought 😅... we can't actually execute the journal after we've written the original value. See my example

Reply:
The second bug should be impossible in theory, If I'm not wrong